compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/calls/TowerLevels.kt

```diff
@@ -190,8 +191,18 @@ class ScopeTowerLevel(
                 empty = false
                 if (candidate.hasConsistentReceivers(extensionReceiver)) {
                     val dispatchReceiverValue =
-                        if (candidate !is FirBackingFieldSymbol) null
-                        else bodyResolveComponents.implicitReceiverStack.lastDispatchReceiver()
+                        when {
+                            candidate !is FirBackingFieldSymbol -> null
+                            candidate.callableId.classId != null -> {
```

Comment:
So, at this point, we know `candidate` is a backing field; `candidate`'s holder information is available. Since receivers on tack are "implicit", I guess we can do further filtering. In this case, since we know the exact holder (of the property), it seems to make sense to me that we should choose the receiver on the stack whose type matches.

Reply:
I think this code is Ok.